src/lib/sed.c

```diff
@@ -186,10 +186,10 @@ int  sed_level0_discovery(struct sed_opal_level0_discovery *discv)
 void sed_deinit(struct sed_device *dev)
 {
 	if (dev != NULL) {
-		curr_if->deinit_fn(dev);
+		if (dev->fd)
```

Comment:
this if check is redundant -- because opal_deinit_pt already ensures that dev->fd is non-zero. please remove it